src/brpc/parallel_channel.cpp

```diff
@@ -139,6 +139,18 @@ class ParallelChannelDone : public google::protobuf::Closure {
             new (d->sub_done(i)) SubDone;
             d->sub_done(i)->cntl.ApplyClientSettings(settings);
             d->sub_done(i)->cntl.allow_done_to_run_in_place();
+
+            // Propagate all HTTP headers from parent controller to sub-controllers.
+            // This preserves application-set headers (e.g., Authorization) on each sub-call.
```

Comment:
Does it need to check if sub-controller headers has exists key before AppendHeader or SetHeader? Avoid overwrite existing header.